pex/resolve/resolver_configuration.py

```diff
@@ -24,7 +25,7 @@
     from pex.third_party import attr
 
 
-PYPI = "https://pypi.org/simple"
+PYPI = os.environ.get("_PEX_TEST_DEFAULT_INDEX", "https://pypi.org/simple")
```

Comment:
There's something a little wrinkly about having an affordance for internal tests in production code. It's not terrible, just awkward. 

Could the tests set a custom index via `--no-pypi --index=...`? Or would that then not be testing some path we care about? Possibly we can test that we correctly default to real PyPI in one place and rely on a custom index elsewhere?

Or maybe in pex3 we can switch to having the pypi url be the default value of `--index` (which is its effect in the code anyway), and thus combine `--pypi` and `--index`, and remove some complexity in the options and here?

Reply:
Agreed, but passing these arguments in hundreds of tests was not attractive. This actually is intertwined with support for pip.conf in #336 & #838, but one thing at a time.